src/sr_test_utils.erl.old

```diff
@@ -0,0 +1,96 @@
+-module(sr_test_utils).
+
+-export([test/1]).
+
+%%==============================================================================
+%% API
+%%==============================================================================
+-spec test(module()) -> ok.
+test(Handler) ->
+  [Trails] = Handler:trails(),
+
+  Path = trails:path_match(Trails),
+  #{get := Get,
+    post := Post} = Metadata = trails:metadata(Trails),
+
+  % This only works with json for now (make a case later)
+  ContentType = ["application/json"],
+  ContentType = maps:get(produces, Get),
+  ContentType = maps:get(consumes, Post),
+  ContentType = maps:get(produces, Post),
+
+  % Create a new element
+  [RequestBody] = maps:get(parameters, Post),
+  PostSchema = maps:get(schema, RequestBody),
+  % For now, we are only adding the required fields
+  Definitions = maps:get(<<"properties">>, PostSchema),
+  RequiredFields = maps:get(<<"required">>, PostSchema),
+  Json = [build_field(F, Definitions) || F <- RequiredFields],
+
+  io:format("Json: ~p~n", [Json]),
+  R = api_call(Post, Path, [{<<"Content-Type">>, ContentType}], maps:from_list(Json)),
```

Comment:
Line 31 is too long:   R = api_call(Post, Path, [{<<"Content-Type">>, ContentType}], maps:from_list(Json)),.